ot/lp/__init__.py

```diff
@@ -656,7 +656,7 @@ def emd_1d(x_a, x_b, a=None, b=None, metric='sqeuclidean', p=1., dense=True,
     perm_a = np.argsort(x_a_1d)
     perm_b = np.argsort(x_b_1d)
 
-    G_sorted, indices, cost = emd_1d_sorted(a, b,
+    G_sorted, indices, cost = emd_1d_sorted(a[perm_a.flatten()], b[perm_b.flatten()],
```

Comment:
I guess you do not have to flatten the permutation indices since they are computed from 1d arrays, or do you ?

Reply:
Could you explain why you flatten here?

Otherwise, LGTM!